src/pip/_internal/index/package_finder.py

```diff
@@ -512,14 +517,17 @@ def _sort_key(self, candidate):
         if link.is_wheel:
             # can raise InvalidWheelFilename
             wheel = Wheel(link.filename)
-            if not wheel.supported(valid_tags):
+            if self._prefer_binary:
+                binary_preference = 1
```

Comment:
Does moving this earlier have any benefits? It feels like unnecessary churn to this code and PR.

Reply:
I think part of why this looked weird is that the prefer_binary bit was splitting the "is_supported" and "get_priority" which were merged into a single call.